ptsa/extensions/edf/edffile.cpp

```diff
@@ -162,6 +162,22 @@ class EDFFile
         return this->header.signalparam[channel].smp_in_file;
     }
 
+    /**
+     * Return the sampling rate of the recording. It is generally assumed that all channels
+     * will be recorded at the same sampling rate, but you can also specify a
+     * specific channel.
+     * @param channel - channel number
+     **/
+    inline auto get_samplerate(int channel)
+    {
+        if (channel >= this->header.edfsignals)
+        {
+            const auto msg = "Channl " + std::to_string(channel) + " out of range";
+            throw py::index_error(msg);
+        }
+        return ((double)this->header.signalparam[channel].smp_in_datarecord / (double)this->header.datarecord_duration) * EDFLIB_TIME_DIMENSION;
```

Comment:
Casting should be done using `static_cast`:

```c++
return static_cast<double>(this->header.signalparam[channel].smp_in_datarecord)
    / static_cast<double>(this->header.datarecord_duration)) * EDFLIB_TIME_DIMENSION;
```

See [this](https://github.com/lefticus/cppbestpractices/blob/master/04-Considering_Safety.md#use-c-style-cast-instead-of-c-style-cast) for details.

Reply:
This change has now been made